aiida/tools/importexport/dbexport/__init__.py

```diff
@@ -167,113 +211,142 @@ def export(
+    entities: Optional[Iterable[Any]] = None,
+    filename: Optional[str] = None,
+    use_compression: bool = True,
+    **traversal_rules: bool,
+) -> Tuple[float, ...]:
     """Export in a zipped folder
 
     .. deprecated:: 1.2.1
         Support for the parameters `what` and `outfile` will be removed in `v2.0.0`.
         Please use `entities` and `filename` instead, respectively.
 
     :param entities: a list of entity instances; they can belong to different models/entities.
-    :type entities: list
 
-    :param filename: the filename (possibly including the absolute path) of the file on which to export.
-    :type filename: str
+    :param filename: the filename
+        (possibly including the absolute path) of the file on which to export.
 
     :param use_compression: Whether or not to compress the zip file.
-    :type use_compression: bool
+
     """
     # Backwards-compatibility
-    entities = deprecated_parameters(
-        old={
-            'name': 'what',
-            'value': kwargs.pop('what', None)
-        },
-        new={
-            'name': 'entities',
-            'value': entities
-        },
+    entities = cast(
+        Iterable[Any],
+        deprecated_parameters(
+            old={
+                'name': 'what',
+                'value': traversal_rules.pop('what', None)
+            },
+            new={
+                'name': 'entities',
+                'value': entities
+            },
+        ),
     )
-    filename = deprecated_parameters(
-        old={
-            'name': 'outfile',
-            'value': kwargs.pop('outfile', None)
-        },
-        new={
-            'name': 'filename',
-            'value': filename
-        },
+    filename = cast(
+        str,
+        deprecated_parameters(
+            old={
+                'name': 'outfile',
+                'value': traversal_rules.pop('outfile', None)
+            },
+            new={
+                'name': 'filename',
+                'value': filename
+            },
+        ),
     )
 
-    type_check(entities, (list, tuple, set), msg='`entities` must be specified and given as a list of AiiDA entities')
+    type_check(
+        entities,
+        (list, tuple, set),
+        msg='`entities` must be specified and given as a list of AiiDA entities',
+    )
     entities = list(entities)
 
     if type_check(filename, str, allow_none=True) is None:
         filename = 'export_data.aiida'
 
     with ZipFolder(filename, mode='w', use_compression=use_compression) as folder:
         time_start = time.time()
-        export_tree(entities=entities, folder=folder, **kwargs)
+        export_tree(entities=entities, folder=folder, **traversal_rules)
         time_end = time.time()
 
     return (time_start, time_end)
 
 
-def export_tar(entities=None, filename=None, **kwargs):
+def export_tar(
+    entities: Optional[Iterable[Any]] = None,
+    filename: Optional[str] = None,
+    **traversal_rules: bool,
+) -> Tuple[float, ...]:
```

Comment:
At least to me, this function doesn't seem to return anything (?)

Reply:
see below...

Reply:
ah I can't comment on it, but its line 357

```python
return (time_export_start, time_export_end, time_compress_start, time_compress_end)
```

Reply:
ok!